File: output/Qwen_repaired_1/avatar/Python/Java/codeforces_637_A.java
import java.util.*;
import java.io.*;

public class codeforces_637_A {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(reader.readLine());
        String[] items = reader.readLine().split(" ");
        Map<String, Integer> counts = new HashMap<>();
        for (String item : items) {
            counts.put(item, counts.getOrDefault(item, 0) + 1);
        }
        List<String> sortedItems = Arrays.asList(items);
        sortedItems.sort(null);
        String result = "";
        int maxCount = -1;
        for (Map.Entry<String, Integer> entry : counts.entrySet()) {
            if (entry.getValue() == counts.values().stream().max(Integer::compare).orElse(-1)) {
                int index = sortedItems.indexOf(entry.getKey());
                if (index > maxCount) {
                    result = entry.getKey();
                    maxCount = index;
                }
            }
        }
        System.out.println(result);
    }
}

File: output/Llama-2-100/avatar/Java/Python/codeforces_637_A.py
def codeforces_637_A ( n ):

sc = input ()
pw = output ()

ara = [ int ( x ) for x in sc.split ( ) ]

map = { x : 1 for x in ara }

for x in ara :

if x not in map :

map [ x ] = 1

else :

map [ x ] += 1

max = 0

for k, v in map.items ( ):

if v == max :

print ( k, end = " " )

break

else :

map [ k ] += 1

pw.close ( )

def debug ( * obj ):

print ( obj )

n = int ( input () )

ara = [ int ( x ) for x in input ().split ( ) ]

codeforces_637_A ( n )

debug ( "ara", ara )

debug ( "map", map )

debug ( "mm", mm )
